js/modules/loki_file_server_api.js

```diff
@@ -43,6 +53,182 @@ class LokiFileServerAPI {
     return this._setOurDeviceMapping(authorisations, isPrimary);
   }
 
+  async getDeviceMappingForUsers(pubKeys) {
+    const users = await this._server.getUsers(pubKeys);
+    return users;
+  }
+
+  async verifyUserObjectDeviceMap(pubKeys, isRequest, iterator) {
+    const users = await this.getDeviceMappingForUsers(pubKeys);
+    // log.info('verifyUserObjectDeviceMap Found', users.length, 'users')
+
+    // go through each user and find deviceMap annotations
+    const notFoundUsers = [];
+    users.forEach(user => {
+      let found = false;
+      if (!user.annotations || !user.annotations.length) {
+        log.info(
+          `verifyUserObjectDeviceMap no annotation for ${user.username}`
+        );
+        return;
+      }
+      user.annotations.forEach(note => {
+        if (note.type !== 'network.loki.messenger.devicemapping') {
+          return;
+        }
+        // isn't desired type
+        // request is slave => primary type...
+        if (
+          (isRequest && note.value.isPrimary !== '0') ||
+          (!isRequest && note.value.isPrimary === '0')
+        ) {
+          /* log.info(`verifyUserObjectDeviceMap found wrong type of` +
+                  `relationship ${user.username}`); */
+          // console.log(`https://file.lokinet.org/users/@${user.username}?prettyPrint=1&include_annotations=1`);
+          return;
+        }
+        const { authorisations } = note.value;
+        if (!Array.isArray(authorisations)) {
+          return;
+        }
+        authorisations.forEach(auth => {
+          // log.info('devmap auth', auth);
```

Comment:
Actually can you check if you could use
```js
const valid = await libloki.crypto.validateAuthorisation(auth);
```
this should already do all the necessary checks :)

Reply:
I'm concerned about the `ourPubKey` behavior in that function (looking at #430) but haven't thought through all the various branches...